File: Examples/Features/Charts/ChartGallery/PieChart.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace GrapeCity.Documents.Excel.Examples.Features.Charts.ChartGallery
{
    public class PieChart : ExampleBase
    {
        public override void Execute(GrapeCity.Documents.Excel.Workbook workbook)
        {
            IWorksheet worksheet = workbook.Worksheets[0];
            GrapeCity.Documents.Excel.Drawing.IShape shape = worksheet.Shapes.AddChart(GrapeCity.Documents.Excel.Drawing.ChartType.Pie, 250, 20, 360, 230);
            worksheet.Range["A1:B4"].Value = new object[,] {
                { "Blue", 1 },
                { "Red", 2 },
                { "Green", 3 },
                { "Purple", 4 },             
            };
            shape.Chart.SeriesCollection.Add(worksheet.Range["A1:B4"], GrapeCity.Documents.Excel.Drawing.RowCol.Columns);
            shape.Chart.ChartTitle.Text = "Pie Chart";
            shape.Chart.Legend.Position = GrapeCity.Documents.Excel.Drawing.LegendPosition.Right;
        }
    }
}

File: Examples.VB/Features/Charts/ChartGallery/PieChart.vb
Namespace Features.Charts.ChartGallery
    Public Class PieChart
        Inherits ExampleBase
        Public Overrides Sub Execute(workbook As Excel.Workbook)
            Dim worksheet As IWorksheet = workbook.Worksheets(0)

            Dim shape As IShape = worksheet.Shapes.AddChart(ChartType.Pie, 250, 20, 360, 230)
            worksheet.Range("A1:B4").Value = New Object(,)  {
                {"Blue", 1},
                {"Red", 2},
                {"Green", 3},
                {"Purple", 4}
            }

            shape.Chart.SeriesCollection.Add(worksheet.Range("A1:B4"), RowCol.Columns)
            shape.Chart.ChartTitle.Text = "Pie Chart"
            shape.Chart.Legend.Position = LegendPosition.Right
        End Sub
    End Class
End Namespace
